client/app/lib/redux/modules/sidebar/stacks.coffee

```diff
@@ -0,0 +1,306 @@
+  myStacks
+  draftStackTemplates
+  (stacks, templates) ->
+    return null  unless stacks and templates
+
+    _.merge stacks, templates
+)
+
+stacksAndMachines = createSelector(
+  sidebarStacks
+  bongo.all 'JMachine'
+  (stacks, machines) ->
+    return null  unless stacks and machines
+
+    return _.mapValues stacks, (stack, key) ->
+      return stack.machines
+        .map (machineId) -> machines[machineId]
+        .filter Boolean
+)
+
+stacksAndTemplates = createSelector(
+  sidebarStacks
+  myStackTemplates
+  (stacks, templates) ->
+    return null unless stacks and templates
+
+    return _.mapValues stacks, (stack, key) ->
+      if stack.baseStackId
+        templates[stack.baseStackId]
+      else stack
+)
+
+stacksAndCredential = createSelector(
+  sidebarStacks
+  bongo.all 'JCredential'
+  (stacks, credentials) ->
+    return _.mapValues stacks, (stack, key) ->
+      provider = stackProvider stack
+      return _.values(credentials)
+        .filter (credential) ->
+          stack.credentials?["#{provider}"]?.first is credential.identifier
+)
+
+stackProvider = (stack) ->
+
+  { config, credentials } = stack
+  return 'managedVM'  unless config
+  { requiredProviders } = config
+  for selectedProvider in requiredProviders
+    break  if selectedProvider in ['aws', 'vagrant']
+
+  selectedProvider ?= (Object.keys credentials ? { aws: yes }).first
+  selectedProvider ?= 'aws'
+
+  return selectedProvider
+
+
+stacksRevisionStatus = (state) ->
+
+  state.stacksAndRevisions
+
+
+stacksAndMenuItems = createSelector(
+  sidebarStacks
+  stacksRevisionStatus
+  (stacks, status) ->
+
+    return unless stacks and status
+
+    return _.mapValues stacks, (stack, key) ->
+      menuItems = {}
+
+      if stack.bongo_.constructorName is 'JComputeStack'
+        revision = status?[stack._id]
+        if revision?.status?.code
+          menuItems['Update'] = null
+
+        managedVM = stack.title.indexOf('Managed VMs') > -1
+
+        if managedVM
+          menuItems['VMs'] = null
+        else
+          menuItems['Edit'] = null  if isAdmin() and not stack.config.oldOwner
+          menuItems['Reinitialize'] = null  unless stack.config.oldOwner
+          ['VMs', 'Destroy VMs'].forEach (name) ->
+            menuItems[name] = null
+          if isAdmin() and not isStackTemplateSharedWithTeam stack.baseStackId
+            menuItems['Make Team Default'] = null
+        return menuItems
+      else
+        ['Edit', 'Initialize'].forEach (name) -> menuItems[name] = null
+        menuItems['Make Team Default'] = null  if isAdmin() and not stack.baseStackId
+        return menuItems
+)
+
+
+sharedVMs = createSelector(
+  bongo.all 'JMachine'
+  (machines) ->
+
+    _.values(machines).filter (machine) ->
+      return no  unless machine.users.length > 1
+
+      # dont show sharedVMs to its owner
+      { profile: { nickname } } = whoami()
+      ownerNickname = getMachineOwner machine, yes
+      nickname isnt ownerNickname
+)
```

Comment:
Ideally we want to use `bongo.byId 'JAccount', whoami()._id` here, but since `whoami` already has the data, it makes sense to use it here. 

One thing about that, the instance returned from `whoami` might not be the actual `JAccount` instance but rather a plain object.

Right now it's ok, but let's find a better solution without requiring to use additional utils for these kind of stuff.